File: GAMECARO/src/FormLogin.cpp
#include "Forms/FormLogin.h"
#include "Forms/FormMain.h"

using namespace Client::Forms;
using namespace System::Windows::Forms;
using namespace System::Drawing;

FormLogin::FormLogin()
{
    InitializeComponent();

    // ==== FORM STYLE ====
    this->Text = "Đăng nhập trước khi chơi";
    this->StartPosition = FormStartPosition::CenterScreen;
    this->ClientSize = Size(420, 220);
    this->BackColor = Color::WhiteSmoke;
    this->FormBorderStyle = FormBorderStyle::FixedSingle;
    this->MaximizeBox = false;

    // ==== TITLE ====
    Label^ lblTitle = gcnew Label();
    lblTitle->Text = "Chào mừng bạn đến với Caro Game";
    lblTitle->Font = gcnew Font("Segoe UI", 14, FontStyle::Bold);
    lblTitle->ForeColor = Color::DarkSlateBlue;
    lblTitle->AutoSize = false;
    lblTitle->TextAlign = ContentAlignment::MiddleCenter;
    lblTitle->Dock = DockStyle::Top;
    lblTitle->Height = 50;
    this->Controls->Add(lblTitle);

    // ==== LABEL NHẬP TÊN ====
    Label^ lbl = gcnew Label();
    lbl->Text = "Nhập tên của bạn:";
    lbl->Left = 30;
    lbl->Top = 70;
    lbl->AutoSize = true;
    lbl->Font = gcnew Font("Segoe UI", 11);
    this->Controls->Add(lbl);

    // ==== TEXTBOX ====
    txtName = gcnew TextBox();
    txtName->Left = 30;
    txtName->Top = 100;
    txtName->Width = 350;
    txtName->Font = gcnew Font("Segoe UI", 11);
    this->Controls->Add(txtName);

    // ==== BUTTON ====
    Button^ btn = gcnew Button();
    btn->Text = "Play";
    btn->Left = 30;
    btn->Top = 140;
    btn->Width = 350;
    btn->Height = 40;
    btn->BackColor = Color::MediumSlateBlue;
    btn->ForeColor = Color::White;
    btn->Font = gcnew Font("Segoe UI", 11, FontStyle::Bold);
    btn->FlatStyle = FlatStyle::Flat;
    btn->FlatAppearance->BorderSize = 0;
    this->Controls->Add(btn);

    btn->Click += gcnew EventHandler(this, &FormLogin::BtnPlay_Click);
}

void FormLogin::BtnPlay_Click(Object^ sender, EventArgs^ e)
{
    String^ name = txtName->Text->Trim();
    if (String::IsNullOrWhiteSpace(name))
    {
        MessageBox::Show("Tên không được để trống!");
        return;
    }

    FormMain^ fm = gcnew FormMain(name);
    fm->Show();
    this->Hide();
}

File: GAMECARO/Forms/FormLogin.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace Client.Forms
{
    public partial class FormLogin : Form
    {
        public FormLogin()
        {
            InitializeComponent();

            // ==== FORM STYLE ====
            this.Text = "Đăng nhập trước khi chơi";
            this.StartPosition = FormStartPosition.CenterScreen;
            this.ClientSize = new Size(420, 220);
            this.BackColor = Color.WhiteSmoke;
            this.FormBorderStyle = FormBorderStyle.FixedSingle;
            this.MaximizeBox = false;

            // ==== TITLE ====
            Label lblTitle = new Label()
            {
                Text = "Chào mừng bạn đến với Caro Game",
                Font = new Font("Segoe UI", 14, FontStyle.Bold),
                ForeColor = Color.DarkSlateBlue,
                AutoSize = false,
                TextAlign = ContentAlignment.MiddleCenter,
                Dock = DockStyle.Top,
                Height = 50
            };
            this.Controls.Add(lblTitle);

            // ==== LABEL NHẬP TÊN ====
            Label lbl = new Label()
            {
                Text = "Nhập tên của bạn:",
                Left = 30,
                Top = 70,
                AutoSize = true,
                Font = new Font("Segoe UI", 11)
            };
            this.Controls.Add(lbl);

            // ==== TEXTBOX ====
            TextBox txt = new TextBox()
            {
                Name = "txtName",
                Left = 30,
                Top = 100,
                Width = 350,
                Font = new Font("Segoe UI", 11)
            };
            this.Controls.Add(txt);

            // ==== BUTTON ====
            Button btn = new Button()
            {
                Text = "Play",
                Left = 30,
                Top = 140,
                Width = 350,
                Height = 40,
                BackColor = Color.MediumSlateBlue,
                ForeColor = Color.White,
                Font = new Font("Segoe UI", 11, FontStyle.Bold),
                FlatStyle = FlatStyle.Flat
            };
            btn.FlatAppearance.BorderSize = 0;
            this.Controls.Add(btn);

            // ==== SỰ KIỆN CLICK ====
            btn.Click += (s, e) =>
            {
                string name = txt.Text.Trim();

                if (string.IsNullOrWhiteSpace(name))
                {
                    MessageBox.Show("Tên không được để trống!");
                    return;
                }

                FormMain fm = new FormMain(name);
                fm.Show();
                this.Hide();
            };
        }
    }
}
